rust/rbac-registration/src/registration/cardano/mod.rs

```diff
@@ -316,23 +392,21 @@ impl RegistrationChainInner {
             return None;
         };
 
-        let point_tx_idx = cip509.origin().clone();
-        let current_tx_id_hash = PointData::new(point_tx_idx.clone(), cip509.txn_hash());
-        let validation_signature = cip509.validation_signature().cloned();
-        let raw_aux_data = cip509.raw_aux_data().to_vec();
+        let Some(registration) = cip509.metadata().cloned() else {
```

Comment:
Is `cloned()` needed here? I can't immediately see why.

```suggestion
        let Some(registration) = cip509.metadata() else {
```